tensorflow/training/docker/2.11/py3/Dockerfile.cpu

```diff
@@ -230,90 +231,89 @@ CMD ["/bin/bash"]
-#     && $PYTHON setup.py install && cd .. && rm -rf sagemaker-debugger
-
-# # install sagemaker-tensorflow from source
-# ARG SAGEMAKER_TENSORFLOW_TAG=v1.16.0
-# RUN apt update && apt install -y libcurl4-openssl-dev
-# RUN git clone -b tf-2 https://github.com/aws/sagemaker-tensorflow-extensions.git \
-#     && cd sagemaker-tensorflow-extensions \
-#     && git checkout tags/$SAGEMAKER_TENSORFLOW_TAG -b $SAGEMAKER_TENSORFLOW_TAG \
-#     && $PYTHON -m pip install . \
-#     && cd .. && rm -rf sagemaker-tensorflow-extensions
+FROM common AS sagemaker
+
+LABEL maintainer="Amazon AI"
+LABEL dlc_major_version="1"
+
+# sagemaker-specific environment variable
+ENV SAGEMAKER_TRAINING_MODULE sagemaker_tensorflow_container.training:main
+
+# dependencies for opencv
+# these dependencies are not needed for gpu image
+RUN apt-get update && apt-get install -y --no-install-recommends \
+    libgtk2.0-dev \
+    && apt-get install -y -qq libkrb5-dev \
+    && apt-get install -y -qq libsasl2-dev libsasl2-modules \
+    && apt-get install -y -qq krb5-user \
+&& rm -rf /var/lib/apt/lists/*
+
+# SMDEBUGGER TESTTING PLEASE REMOVE
+ARG TF_URL=https://framework-binaries.s3.us-west-2.amazonaws.com/tensorflow/r2.11_aws/cpu/2022-12-22-20-46/tensorflow_cpu-2.11.0-cp39-cp39-manylinux_2_17_x86_64.manylinux2014_x86_64.whl
+# https://github.com/tensorflow/models/issues/9267
+# tf-models does not respect existing installations of TF and always installs open source TF
+RUN ${PIP} install --no-cache-dir -U \
+    tf-models-official==2.11.0 
+
+RUN ${PIP} install --no-cache-dir -U \
+    tensorflow-text==2.11.0
+
+RUN ${PIP} uninstall -y tensorflow tensorflow-gpu \
+    ; ${PIP} install --no-cache-dir -U \
+    ${TF_URL} \
+    tensorflow-io==0.27.0 \
+    tensorflow-datasets==4.7.0
+    
+# Install Horovod with TensorFlow
+RUN HOROVOD_WITH_MPI=1 \
+   && HOROVOD_WITH_TENSORFLOW=1 HOROVOD_WITHOUT_MXNET=1 HOROVOD_WITHOUT_PYTORCH=1 \
+   && ${PIP} install --no-cache-dir horovod==0.26.1
+
+RUN $PYTHON -m pip install --no-cache-dir -U \
+    "bokeh>=2.3,<3" \
+    "imageio>=2.9,<3" \
+    "opencv-python>=4.3,<5" \
+    "plotly>=5.1,<6" \
+    "seaborn>=0.11,<1" \
+    "numba<0.54" \
+    "shap>=0.39,<1"
+
+RUN $PYTHON -m pip install --no-cache-dir -U \
+    "sagemaker>=2.93.0,<3" \
+    sagemaker-experiments==0.* \
+    "sagemaker-tensorflow-training>=20.3.1" \
+    "sagemaker-training>=4.2.8" \
+    "sagemaker-studio-analytics-extension==0.0.2" \
+    "sparkmagic<1" \
+    "sagemaker-studio-sparkmagic-lib<1" \
+    smclarify
+
+# install smdebug directly the specific branch
+ARG SMDEBUG_TAG=1.0.24
+RUN git clone https://github.com/awslabs/sagemaker-debugger.git \
+    && cd sagemaker-debugger \
+    && git checkout tags/$SMDEBUG_TAG -b $SMDEBUG_TAG \
+    && $PYTHON setup.py install && cd .. && rm -rf sagemaker-debugger
+
+# install sagemaker-tensorflow from source
+ARG SAGEMAKER_TENSORFLOW_TAG=v1.17.0
+RUN apt update && apt install -y libcurl4-openssl-dev
```

Comment:
Why is `libcurl4-openssl-dev` installed here, and not at the top with all other OS packages?

Also, this `apt update && apt install` needs to be followed by
```
 && rm -rf /var/lib/apt/lists/* \
 && apt-get clean
```

Reply:
Updated!